crm_lead_vat/models/crm_lead.py

```diff
@@ -3,6 +3,11 @@
 
 from odoo import fields, models
 
+from odoo.addons.crm.models.crm_lead import PARTNER_ADDRESS_FIELDS_TO_SYNC
+
+# Add vat in list
+PARTNER_ADDRESS_FIELDS_TO_SYNC += ["vat"]
+
```

Comment:
This code is loaded when the module is scanned by Odoo, at boot.

🛑 Thus you shouldn't alter this global variable (actually a pseudo-constant, as stated by the UPPER_CASE name).